dist.ts

```diff
@@ -0,0 +1,159 @@
+    const keys = compatNode.toJS(yaml) as Array<string>;
+    features = keys.map((key) => compat.feature(key));
+  } else {
+    // If there are no compat keys, get feature data from BCD tags
+    features = lookupTag(id);
+    if (features.length > 0) {
+      yaml.set(
+        "compat_features",
+        features.map((f) => f.id),
+      );
+    }
+    // TODO: Warn if there are no compat keys? Or will that be self-evident?
+  }
+
+  if (!statusNode && features.length) {
+    // Set a status, if one doesn't already exist and there's feature data we can
+    // use.
+
+    const compatKeys = features.map((f) => f.id) as [string, ...string[]];
+    const status = computeBaseline({ compatKeys, checkAncestors: true });
+
+    // Creat the status node and insert it just before "compat_features"
+    statusNode = yaml.createPair("status", JSON.parse(status.toJSON()));
+    assert(YAML.isMap(yaml.contents));
+    const targetIndex = yaml.contents.items.findIndex(
+      (item) => item.key.toString() === "compat_features",
+    );
+    yaml.contents.items.splice(
+      targetIndex,
+      0,
+      statusNode as YAML.Pair<any, any>,
+    );
+  } else {
+    // If there is a support block, then we set its baseline_high_date, if
+    // applicable.
+
+    // TODO: consider checking if a generated status would be the same, and warn
+    // about it.
+
+    // TODO: consider checking if a generated status would be _different_, and
+    // warn about it.
+
+    const baseline = yaml.getIn(["status", "baseline"]) as
+      | "high"
+      | "low"
+      | false;
+    const lowDate = yaml.getIn(["status", "baseline_low_date"]) as
+      | string
+      | undefined;
+    const highDate = yaml.getIn(["status", "baseline_high_date"]) as
+      | string
+      | undefined;
+    if (baseline === "high" && !highDate) {
+      // Append a high date…
+      yaml.setIn(
+        ["status", "baseline_high_date"],
+        Temporal.PlainDate.from(lowDate)
+          .add(BASELINE_LOW_TO_HIGH_DURATION)
+          .toString(),
+      );
+
+      // …then fix the order.
+      assert(YAML.isMap(statusNode));
+      const highDateNode = statusNode.items.pop();
+      const targetIndex = statusNode.items.findIndex(
+        (item) => item.key === "baseline_low_date",
+      );
+      statusNode.items.splice(targetIndex, 0, highDateNode);
+    }
+  }
+
+  return yaml;
+}
+
+const compat = new Compat();
+
+/**
+ * Get all of the features with a given @mdn/browser-compat-data tag.
+ *
+ * @param {string} tag A tag (excluding the `web-features:` prefix).
+ */
+function lookupTag(tag: string): Feature[] {
+  const namespaceTag = `web-features:${tag}`;
+
+  const features = [];
+  for (const feature of compat.walk()) {
+    if (feature.tags.includes(namespaceTag)) {
+      features.push(feature);
+    }
+  }
+  return features;
+}
+
+if (process.argv[1] === fileURLToPath(import.meta.url)) {
+  for (const y of process.argv.slice(2)) {
+    updateDistFile(y);
+  }
+}
```

Comment:
This ought to use the yargs machinery we use elsewhere, for good help messages and such. But I wanted to confirm that we're happy with the overall approach before putting in the work.

Reply:
I am very happy.